### ISSUE TITLE
Error when receiving a event coming from an ALB triggering the Lambda

### ISSUE THREAD

**Issue by AaMadera** [sst/open-next#278]
Hi team, after deploying a lambda using open-next, 
with a newer version than `open-next": "~0.7.0"` 
I found the current error log in CloudWatch:

```sh
 ERROR    [dd.trace_id=520218430 dd.span_id=89384760001517] 
Invoke Error     
{
  "errorType": "TypeError",
  "errorMessage": "Cannot read properties of undefined (reading 'sourceIp')",
  "stack": [
    "TypeError: Cannot read properties of undefined (reading 'sourceIp')",
    "    at E9 (file:///var/task/index.mjs:34:4606)",
    "    at zM (file:///var/task/index.mjs:34:4198)",
    "    at cse (file:///var/task/index.mjs:34:10506)",
    "    at /var/task/node_modules/datadog-lambda-js/dist/utils/handler.js:166:25",
    "    at /var/task/node_modules/datadog-lambda-js/dist/index.js:223:70",
    "    at step (/var/task/node_modules/datadog-lambda-js/dist/index.js:44:23)",
    "    at Object.next (/var/task/node_modules/datadog-lambda-js/dist/index.js:25:53)",
    "    at /var/task/node_modules/datadog-lambda-js/dist/index.js:19:71",
    "    at new Promise (<anonymous>)",
    "    at __awaiter (/var/task/node_modules/datadog-lambda-js/dist/index.js:15:12)"
  ]
}
```
My lambda is triggered by an [ALB](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/introduction.html)

so, is open-next compatible with it?
my guess is that the ALB event doesn't have that expected event structure where you check the `sourceIp`

currently I'm stuck with version `open-next": "~0.7.0"`, because this version allows me to run my lambda,
any newer major version throws the previously mentioned error

Currently I'm using [ALB](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/introduction.html) instead of API-GateWay because at some point started to be the cheaper option for my infra, I think this case happens to more people.

Let me know if you need me to provide more details or resources to help debugging

**Comment by khuezy** [comment]
Hi thanks for the report, we access `sourceIp` in two places:

https://github.com/sst/open-next/blob/main/packages/open-next/src/adapters/event-mapper.ts#L102
https://github.com/sst/open-next/blob/main/packages/open-next/src/adapters/event-mapper.ts#L125

We probably don't handle ALB events correctly: https://github.com/sst/open-next/blob/main/packages/open-next/src/adapters/event-mapper.ts#L67C1-L73C49

Feel free to create a PR if you want to tackle this.

**Comment by conico974** [comment]
This could be fixed in `open-next@3.0.0-rc.1` https://open-next.js.org/v3
You'll have to write a custom converter for the alb event though

**Comment by nicholas-c** [comment]
Little late to the party here, but just adding some context for any other adventurers who get here.

I've been doing some evaluation of using ALB for other purposes and I don't think they're compatible with Next.js or mots React Frameworks moving forward.

Although as @conico974 has said above, a custom event converter is available and easy enough to achieve basic event mapping with ALB's. There is a wider issue that ALB's do not support streaming, this is an absolute killer if you want to use Suspense and React Server Components and maintain a low latency TTFB 🙏

**Comment by AaMadera** [comment]
Hey guys, thank you so much for all your comments
unfortunately we haven't had enough time to review all the options you have mentioned here

but we created a patch to let us use open-next for ALB in this fixed version: 2.3.4

I'm not aware about the compatibility of the react features mentioned above, but for our special case, our app benefits from using ALB because of the amount of requests, it significantly reduced the costs compared to API-GateWay

I feel we will end up using the v3 soon, so we either use the custom converter or create a similar patch as the described here.

Sorry for offering the patch only, and not a PR, the thing is that we haven't had enough time to create unit tests and other requirements for a proper PR

----
Patch: crated using [yarn patch instructions](https://yarnpkg.com/cli/patch)

Add the provided patch inside your project:
```sh
./.yarn/patches/open-next-2_3_4.patch
```

In your package.json edit open-next dependency as follows:
```json
"open-next": "patch:open-next@2.3.4#.yarn/patches/open-next-2_3_4.patch",
```

we based on [Example Application Load Balancer request event provided by aws](https://docs.aws.amazon.com/lambda/latest/dg/services-alb.html)

[open-next-2_3_4.patch](https://github.com/user-attachments/files/16785952/open-next-2_3_4.patch)
